### PR TITLE
Cli and control port

### PR DESCRIPTION
**Pull request by Doralitze**
This pull request implements a basic CLI port for remote management. Furthermore it implements the management of control desk input configuration as its first command.

Further changes:
 * Append creation of control port object in DMXGui
 * Implement message queuing in Network.py

Remaining bugs:
 * [ ] Closing of control port while gui is shutting down is causing block.
 * [ ] There is no async trigger yet to process queued messages, hence a manual update from the Qt thread is required.
 
Features that will be added later:
 * Management of running scene
 * Updating filter parameters
 * Listing filters
 * Updating input column settings
 * listing scenes

### CODE REVIEW

File: src/model/color_hsi.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        self._intensity: confloat(ge=0, le=1) = intensity
+
+    @property
+    def hue(self) -> confloat(ge=0, le=360):
+        """color itself in the form of an angle between [0,360] degrees"""
+        return self._hue
+
+    @property
+    def saturation(self) -> confloat(ge=0, le=1):
+        """color itself in the form of an angle between [0,360] degrees"""
```
Comment (Doralitze) [inline src/model/color_hsi.py:28]: """saturation of the color. Float between (including) zero (100% white, 0% color) and 1 (0% white, 100% color)"""

File: src/model/color_hsi.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        return self._saturation
+
+    @property
+    def intensity(self) -> confloat(ge=0, le=1):
+        """color itself in the form of an angle between [0,360] degrees"""
```
Comment (Doralitze) [inline src/model/color_hsi.py:33]: """Percieved illuminance, float [0, 1]"""

File: src/model/control_desk.py [hunk 3]
```diff
@@ -1,37 +1,39 @@
+# coding=utf-8
+"""models for X-Touch"""
```
Comment (Doralitze) [inline src/model/control_desk.py:2]: Also for other connected devices like extenders or joystick

File: src/model/control_desk.py [hunk 4]
```diff
@@ -1,37 +1,39 @@
...
+from Network import NetworkManager
+from model.color_hsi import ColorHSI
 
 
 def _generate_unique_id() -> str:
     return str(uuid4())
 
 
 class DeskColumn(ABC):
+    """TODO docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:16]: """This class represents a single column inside a bank.

This class should not be instantiated directly. Instead the implementing classes ColorDeskColumn and RawDeskColumn should be used.

TODO: Also implement U,UA and A types.
"""

File: src/model/control_desk.py [hunk 5]
```diff
@@ -58,6 +61,7 @@ def display_name(self, text: str):
 
     @property
     def bottom_display_line_inverted(self) -> bool:
+        """TODO Docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:64]: """True if the text foreground and background pixels are inverted."""

File: src/model/control_desk.py [hunk 6]
```diff
@@ -127,53 +137,69 @@ def encoder_position(self, position: int):
...
-        self._color_h = h
-        self._color_s = s
-        self._color_i = i
+    def color(self, color: ColorHSI):
+        self.color = color
         self.update()
 
 
 class FaderBank:
+    """TODO Docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:161]: """Store a bank page of columns the user can switch through.

Warning: As of the time of this writing, the buttons for scrolling through a bank are not implemented in fish, hence the user can only access up to 8 columns.
"""

File: src/model/control_desk.py [hunk 7]
```diff
@@ -127,53 +137,69 @@ def encoder_position(self, position: int):
...
         msg = proto.Console_pb2.add_fader_bank_set.fader_bank()
         for col in self.columns:
-            msg.cols.extend([col._generate_column_message()])
+            msg.cols.extend([col._generate_column_message()])  # TODO private Methode
         return msg
 
 
 class BankSet:
-
+    """TODO docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:179]: """This class represents a bank set.

A bank set is a set of banks, the user can switch between at a given time. Multiple bank sets can be linked to fish at the same time but only one may be active at the same time. Only the GUI may specify which bank set is active any given time, except for the event that the active bank set will be unlinked. In this case fish will enable the bank set with the next lower index.
"""

File: src/model/control_desk.py [hunk 8]
```diff
@@ -127,53 +137,69 @@ def encoder_position(self, position: int):
...
     _linked_bank_sets = []
 
+    @classmethod
+    def fish_connector(cls) -> NetworkManager:
+        """Connector of the Bank"""
+        return cls._fish_connector
+
+    @classmethod
+    def linked_bank_sets(cls) -> list:
+        """TODO docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:192]: """This method yields the mutable list of bank sets that are currently loaded in fish.

This method should only be used by friend classes.
"""

File: src/model/control_desk.py [hunk 9]
```diff
@@ -127,53 +137,69 @@ def encoder_position(self, position: int):
...
+        return cls._linked_bank_sets
+
+    @classmethod
+    def active_bank_set(cls) -> str:
+        """current bank set"""
+        return cls._active_bank_set
+
     @staticmethod
     def get_linked_bank_sets():
+        """TODO Docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:202]: """This method returns a copy of the linked bank sets, save to be used by non friend classes."""

File: src/model/control_desk.py [hunk 10]
```diff
@@ -247,24 +273,25 @@ def add_bank(self, bank: FaderBank):
...
             return False
         self.active_bank = i
         text = self.description
-        BankSet._seven_seg_data = (str(i % 100) if i > 9 else "0" + str(i)) + text[-10:] + (" " * (10-len(text)))
+        BankSet._seven_seg_data = (str(i % 100) if i > 9 else "0" + str(i)) + text[-10:] + (" " * (10 - len(text)))
         self._send_desk_update_message()
         return True
 
     def link(self) -> bool:
+        """TODO docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:294]: """Load the bank set to fish.

If there was no bank set linked before this bank set might become the active one immediately."""

File: src/model/control_desk.py [hunk 11]
```diff
@@ -284,28 +311,32 @@ def unlink(self) -> bool:
             else:
                 i += 1
         BankSet._fish_connector.send_fader_bank_set_delete_message(self.id)
-        BankSet._linked_bank_sets.pop(found_index)
+        BankSet._linked_bank_sets.pop(found_index)  # TODO found index koente leer sein
         return True
 
     @property
     def is_linked(self) -> bool:
+        """TODO docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:319]: """Returns True if the bank set is loaded to fish."""

File: src/model/control_desk.py [hunk 12]
```diff
@@ -284,28 +311,32 @@ def unlink(self) -> bool:
...
         return self.pushed_to_fish
 
 
 def set_network_manager(network_manager: NetworkManager):
+    """TODO docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:324]: """Set the network manager instance to be used by all bank sets and subsequent banks and columns."""

File: src/model/control_desk.py [hunk 13]
```diff
@@ -284,28 +311,32 @@ def unlink(self) -> bool:
...
     BankSet._fish_connector = network_manager
 
 
 def set_seven_seg_display_content(content: str):
+    """TODO docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:329]: """Set the content of the 7seg displays of connected X-Touch controllers.

Fish will truncate any text longer than 12 chars and only guarantees the correct display of hexadecimal characters due to the limitations of seven segment displays.
"""

File: src/model/control_desk.py [hunk 14]
```diff
@@ -284,28 +311,32 @@ def unlink(self) -> bool:
...
     BankSet._seven_seg_data = content[0:12] + " " * (12 - len(content))
     # TODO send update message
 
 
-def commit_all_banksets():
+def commit_all_bank_sets():
+    """TODO docstring"""
```
Comment (Doralitze) [inline src/model/control_desk.py:335]: """This method calls update on all linked columns.

This is useful in the event of a reconnect to fish as the state of fish is unknown at this point in time.
"""

**Review by Doralitze — COMMENTED** [review]
I could not edit your doc strings immediately, hence I wrote these comments.

**Review by maxkaussow — APPROVED** [review]
passt

**Pull request merged**